### PR TITLE
Tidy up preload/scanner code

### PR DESCRIPTION
**Pull request by daviddrysdale**
Subsumes some of pull #36. 

Probably easiest to review commit-by-commit.

### CODE REVIEW

**Comment by daviddrysdale** [comment]
@FiloSottile, could you confirm you're OK with me pulling over 2 of your commits from PR #36 ?  Thanks.

**Comment by FiloSottile** [comment]
Yep, I'm happy with you submitting these commits on my behalf. 👍

File: preload/preloader/preloader.go [hunk 1]
```diff
@@ -141,18 +116,19 @@ func precertSubmitterJob(ctx context.Context, addedCerts chan<- *preload.AddedCe
...
 func main() {
 	flag.Parse()
 	var sctFileWriter io.Writer
 	var err error
 	if *sctInputFile != "" {
-		sctFileWriter, err = os.Create(*sctInputFile)
+		sctFile, err := os.Create(*sctInputFile)
 		if err != nil {
-			log.Fatal(err)
+			log.Fatalf("Failed to create SCT file: %v", err)
```
Comment (AlCutter) [inline preload/preloader/preloader.go]: nit - these could probably be `Errorf` since they're in `main`

Comment (daviddrysdale) [reply preload/preloader/preloader.go]: Base Golang `log` doesn't have `Exitf` (which is what I believe you meant).

Comment (AlCutter) [reply preload/preloader/preloader.go]: I did indeed... :coffee:

**Pull request merged**